play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/internal/PlayApiService.kt

```diff
@@ -26,8 +25,24 @@ internal abstract class PlayApiService @Inject constructor(
         private val fileOps: FileSystemOperations,
 ) : BuildService<PlayApiService.Params>, OperationCompletionListener, AutoCloseable {
     val publisher by lazy {
-        credentialStream().use {
-            PlayPublisher(it, parameters.appId.get())
+        val useAppDefaultCreds = parameters.useApplicationDefaultCredentials.getOrElse(false)
+        val useExplicitCreds = (parameters.credentials.asFile.orNull != null ||
+                System.getenv(PlayPublisher.CREDENTIAL_ENV_VAR) != null)
+
+        if (useAppDefaultCreds && useExplicitCreds) {
```

Comment:
Let add another if statement that checks for specifying an impersonation without `useAppDefaultCreds`.

Reply:
👍 
The other way to go would be to allow impersonation with the static creds (i.e. not ADC).. It should work just the same but perhaps it's best to not encourage this path...

Reply:
Can one service account impersonate another? If yes, then sure let's allow it but maybe not document it.

Reply:
Yeah it can, but that path hasn't been implemented.. I was pondering whether we should implement it or just not allow it as suggested.. 
I think I'll go with the second one cos (as mentioned) we should encourage the ADC route

Reply:
Sure, I guess if somebody wants to use impersonation with a service account they can send a PR.